### ISSUE TITLE
Comment:Step-Generators-Run

### ISSUE THREAD

**Issue by swelljoe** [feathersjs/feathers-docs#837]
Comment:Step-Generators-Run

On this page, after following the steps, the tutorial app doesn't run for me. It throws the following error:

```bash
/home/joe/src/feathers/feathers-app/node_modules/feathers-authentication-jwt/lib/index.js:58
      throw new Error('You must provide a \'header\' in your authentication configuration or pass one explicitly');
      ^

Error: You must provide a 'header' in your authentication configuration or pass one explicitly
    at Function.jwtAuth (/home/joe/src/feathers/feathers-app/node_modules/feathers-authentication-jwt/lib/index.js:58:13)
    at Function.configure (/home/joe/src/feathers/feathers-app/node_modules/feathers/lib/application.js:150:8)
    at Function.module.exports (/home/joe/src/feathers/feathers-app/src/authentication.js:12:7)
    at Function.configure (/home/joe/src/feathers/feathers-app/node_modules/feathers/lib/application.js:150:8)
    at Object.<anonymous> (/home/joe/src/feathers/feathers-app/src/app.js:44:5)
    at Module._compile (module.js:573:30)
    at Object.Module._extensions..js (module.js:584:10)
    at Module.load (module.js:507:32)
    at tryModuleLoad (module.js:470:12)
    at Function.Module._load (module.js:462:3)
```

This is caused by the default.json containing the following:

```JSON
    "jwt": {
      "header": {
         "type": "access"
      }
```

But, feathers-authentication-jwt expects `header` to be a string. Changing default.json to contain a string in that field (`"header": "access"`) allows it to start without error. I don't know if I've ended up with the wrong version of the module or if the generator is doing the wrong thing, or what. But, following the docs didn't work, so I guess something somewhere is wrong. I'm too much of an amateur at all of this to guess what/where.

My versions from package.json:

```JSON
 "dependencies": {
    "body-parser": "^1.18.0",
    "compression": "^1.7.0",
    "cors": "^2.8.4",
    "feathers": "^2.2.0",
    "feathers-authentication": "^1.2.7",
    "feathers-authentication-hooks": "^0.1.4",
    "feathers-authentication-jwt": "^0.3.2",
    "feathers-authentication-local": "^0.4.4",
    "feathers-configuration": "^0.4.1",
    "feathers-errors": "^2.9.1",
    "feathers-hooks": "^2.0.2",
    "feathers-hooks-common": "^3.7.2",
    "feathers-nedb": "^2.7.0",
    "feathers-rest": "^1.8.0",
    "feathers-socketio": "^2.0.0",
    "helmet": "^3.8.1",
    "nedb": "^1.8.0",
    "serve-favicon": "^2.4.3",
    "winston": "^2.3.1"
  },
```

**Comment by eddyystop** [comment]
The error is thrown after exactly which step?

**Comment by josephrace** [comment]
Looks like this is the same issue that I reported here - https://github.com/feathersjs/generator-feathers/issues/278

And someone else reported here - https://github.com/feathersjs/feathers-authentication-jwt/issues/34

**Comment by RobIsHere** [comment]
If you change the "header" to a string e.g.: "header": "Authorization",
you cannot generate JWTs anymore. Login with local strategy fails saying that header needs to be an object.

**Comment by RobIsHere** [comment]
Referring to https://docs.feathersjs.com/api/authentication/server.html#default-options
there shoud be 2 "header" definitions:
- One at top level as a string (This one is missing with feathers generate auth...)
- One at the jwt object as an object

That is quite confusing: 
- The top-level one is the name of the http header in the request containing the auth token. 
- The jwt level one is the JWT-header (maleformed header is in the docs, see https://tools.ietf.org/html/rfc7519#section-5 for standards as header has e.g. no "type" but "typ" etc)

By adding the top-level "header": "Authorization" entry, the application boots, but fails later when calling create at the authenticate service for loging in with username and password. The JWT creation fails saying: "header" must be an object.

**Comment by RobIsHere** [comment]
Hurrah! Made it work! The server starts and the jwt is generated.

Auto generated config/default.json looks like (I let out the other keys):

```
  "authentication": {
    "jwt": {
      "header": { "type": "access" },
    },
```

I've changed that to:

```
  "authentication": {
    "header": "Authorization",  
    "jwt": {
    },
```
(remove header from jwt!)

Although that works, I really don't know if this is a bug in generator and docs or in the code. But for now it's good for me.

**Comment by eddyystop** [comment]
Culd you post an issue in feathers-generator (reference the issue here) to get the proper eyes on it?

**Comment by josephrace** [comment]
I posted a link to this discussion on my ticket there - https://github.com/feathersjs/generator-feathers/issues/278 - and links to other issues people have opened that appear to be the same.

Seems to be some confusion about where the fault lies...

**Comment by daffl** [comment]
Fixed in `feathers-cli@2.3.7`. The error can be fixed in a generated application by changing

```js
app.configure(jwt(config.jwt));
app.configure(local(config.local));
```

To

```js
app.configure(jwt());
app.configure(local());
```
